compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -1667,6 +1667,49 @@ dataDependencyTests Tools{damlc,repl,validate,davlDar,oldProjDar} = testGroup "D
         withCurrentDirectory (tmpDir </> "proj") $
             callProcessSilent damlc ["build"]
 
+    , testCaseSteps "Empty variant constructors" $ \step -> withTempDir $ \tmpDir -> do
+        -- This test checks that variant constructors without argument
+        -- are preserved. This is a regression test for issue #7207.
+        step "building project with type definition"
+        createDirectoryIfMissing True (tmpDir </> "type")
+        writeFileUTF8 (tmpDir </> "type" </> "daml.yaml") $ unlines
+            [ "sdk-version: " <> sdkVersion
+            , "name: type"
+            , "source: ."
+            , "version: 0.1.0"
+            , "dependencies: [daml-prim, daml-stdlib]"
+            ]
+        writeFileUTF8 (tmpDir </> "type" </> "Foo.daml") $ unlines
+            [ "module Foo where"
+            , "data A = B | C Int"
```

Comment:
Should we add a third case `D ()` and check that it behaves like `D` when imported as data-dependency?

Reply:
I don't think so. `D ()` is something we're deprecating, and the fact that it's treated the same as `D` in the current compiler is mostly a mistake.

Reply:
I generally agree but would feel more comfortable to have a test as long as users can still write it, even though they get warned about it.

Reply:
I added a test for `D ()` in the single constructor case, which is something we had talked about adding as a special case  (since we can't distinguish single-constructor variants from enums any other way, i.e. there's no ambiguity). (This actually also came up in one of the other tests, but better to add one here as well.)